client/app/services/resizeObserver.js

```diff
@@ -0,0 +1,31 @@
+const items = new Map();
+
+function checkItems() {
+  items.forEach((item, node) => {
+    const bounds = node.getBoundingClientRect();
+    // convert to int (because these numbers needed for comparisons), but preserve 1 decimal point
+    const width = Math.round(bounds.width * 10);
+    const height = Math.round(bounds.height * 10);
+
+    if (
+      (item.width !== width) ||
+      (item.height !== height)
+    ) {
+      item.width = width;
+      item.height = height;
+      item.callback(node);
+    }
+  });
+
+  setTimeout(checkItems, 100);
+}
+
+checkItems(); // ensure it was called only once!
```

Comment:
@kravets-levko are you sure having this here won't cause any trouble? (e.g. won't run if not needed, would run only once)

Reply:
We discussed it some time ago with @arikfr (he had the same questions). Currently I left this code almost unchanged (just moved it to service to be able to use from React), maybe later I'll fix it to be more clear.

Answering your questions: this function runs only once (because `require` will cache module) and then it will re-schedule itself every 100ms. There are no performance issues (when elements list is empty - this function returns in a millisecond).